R/bin_spatial.R

```diff
@@ -0,0 +1,220 @@
+#' `dist` which represents the distance between the centroid of adjacent cells.
+#' Using this distance, the function identifies which resolution is most
+#' similar to the input `dist`, and uses this resolution.
+#' - Distance: The "dist" method identifies unique localities in the input
+#' `occdf` (i.e., unique pairs of coordinates) and generates a spatial sample
+#' for each locality based on a user-defined distance. All occurrences within
+#' the specified distance from the reference locality are drawn as a single
+#' spatial sample. As high density of occurrences might result in numerous
+#' samples for the same area, the `buffer` argument has been implemented to
+#' allow users to generalise the definition of
+#' a unique locality to incorporate a broader geographic area. The
+#' functionality of this approach is heavily
+#' dependent on the
+#' \code{\link[sf:st_is_within_distance]{sf::st_is_within_distance()}} function.
+#'
+#' Note: prior to implementation of either method, the coordinate reference
+#' system (CRS) for input data is defined as EPSG:4326 (World Geodetic System
+#' 1984). The user might wish to update their data accordingly if this is
+#' not appropriate.
+#'
+#' @section Developer(s):
+#' Lewis A. Jones
+#' @section Reviewer(s):
+#' To be reviewed
+#' @importFrom sf st_as_sf st_is_within_distance st_drop_geometry
+#' @importFrom h3jsr point_to_h3 h3_to_point
+#' @examples
+#' # Get internal data
+#' data("tetrapods")
+#' # Smaller dataframe for examples
+#' occdf <- tetrapods[1:500, ]
+#' # Bin data using a hexagonal equal-area grid
+#' bin_spatial(occdf = occdf, method = "grid", dist = 250)
+#' # Bin data based on distance for each unique location
+#' bin_spatial(occdf = occdf, method = "dist", dist = 250)
+#' # Bin data based on distance with a buffer to define each unique location
+#' bin_spatial(occdf = occdf, method = "dist", dist = 250, buffer = 100)
+#' @export
+bin_spatial <- function(occdf,
+                        lng = "lng",
+                        lat = "lat",
+                        method = "grid",
+                        dist = 250,
+                        buffer = NULL,
+                        return = FALSE) {
+
+  #=== Error handling ===
+  if (!is.data.frame(occdf)) {
+    stop("occdf should be of class dataframe")
+  }
+
+  if (lng %in% colnames(occdf) == FALSE ||
+      lat %in% colnames(occdf) == FALSE) {
+    stop("input column names do not exist in `occdf")
+  }
+
+  if (!is.numeric(occdf[, lng]) || !is.numeric(occdf[, lat])) {
+    stop("input coordinates are not of class numeric")
+  }
+
+  if (any(occdf[, lat] > 90) || any(occdf[, lat] < -90)) {
+    stop("Latitudinal coordinates should be more than -90 and less than 90")
+  }
+
+  if (any(occdf[, lng] > 180) || any(occdf[, lng] < -180)) {
+    stop("Longitudinal coordinates should be more than -180 and less than 180")
+  }
+
+  if (!is.character(method)) {
+    stop("`method` should be of class character")
```

Comment:
I think this error is redundant given the following one? Not that it really matters if it is!

Reply:
Dropped, resolving.